docs/modules/ROOT/pages/adr/42-backup-encryption-for-mariadb-galera-and-postgresql.adoc

```diff
@@ -0,0 +1,133 @@
+
+[source,toml]
+----
+[minio]
+type = s3
+provider = Minio
+env_auth = false
+access_key_id = minioadmin
+secret_access_key = minioadmin
+endpoint = http://127.0.0.1:9000
+region = us-east-1
+location_constraint = us-east-1
+force_path_style = true
+
+[s3-crypt]
+type = crypt
+remote = minio:backups <1>
+filename_encryption = standard
+directory_name_encryption = true
+password = IRv0dJ285I6uxOAKW7tpwirL9TxiF5wbwAlSHA <2>
+password2 = rPRcLTS8YCbzeV_WiDfrB-fhf5f1dGJ94CXBzw <3>
+----
+<1> Use the minio remote as the storage backend
+<2> Password used for encryption
+<3> Password used for salt
+
+We can then run rclone with: `rclone serve s3 s3-crypt: --addr :9090 --auth-key proxyaccess,proxysecret`.
+
+Rclone is the most viable option: it's actively maintained, widely used, and has straightforward configuration.
+
+==== References
+
+- https://rclone.org/crypt/[Rclone crypt]
+- https://rclone.org/commands/rclone_serve_s3/[Rclone serve s3]
+- https://rclone.org/commands/rclone_serve_s3/[Rclone s3 backend]
+
+
+[NOTE]
+====
+**Key rotation**
+
+Since rclone uses the provided password to generate the encryption key, changing the password is not possible for already encrypted data. The data would need to be wiped from the remote and re-encrypted with the new key. This is not feasible in our setup. Since our setup with k8up also doesn't do key rotation, this isn't an issue.
+====
+
+==== Storing encryption passwords
+
+Since we don't plan to rotate the password used for the encryption key, the password can be saved inside of a Kubernetes secret. This is also how we do it with the repository password for K8up.
+
+== Using Rclone with CNPG and MariaDB
+
+Rclone will run as a separate deployment inside each instance namespace, acting as an encryption proxy between the database operators and the object storage backend.
+
+=== Deployment Architecture
+
+The rclone proxy will:
+
+* Deploy as a dedicated service in the instance namespace
+* Connect directly to the provisioned `xobjectbucket` resource
+* Expose an S3-compatible endpoint for the database operators
+
+=== Integration Points
+
+Each database operator integrates with the rclone proxy through its native S3 backup configuration.
+
+**CloudNativePG (CNPG)**
+
+The Barman object storage configuration will point to the rclone service endpoint as its S3 remote, ensuring all PostgreSQL backups are transparently encrypted before reaching the storage backend.
+
+**MariaDB Operator**
+
+The physical backup resource will configure the rclone service endpoint as its target, providing the same transparent encryption for MariaDB Galera backups.
+
+== Decision
+
+We will use rclone as an S3 encryption proxy to provide encrypted backups for MariaDB Galera and PostgreSQL services. Rclone will be deployed as a separate service in each instance namespace, configured with crypt remotes to transparently encrypt all backup data before it reaches the object storage backend.
```

Comment:
Have you created a little PoC to test backup and restore with CNPG or Mariadb using the rclone solution?

Reply:
+1 to that from my side. Also testing the throughput with various resource settings (especially cpu limits) should be done in the PoC as well